src/containers/app/EditApp.js

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { FormControl, FormGroup, ControlLabel, Button, Alert } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import AsyncContent, { AsyncStatePropType } from '../../components/asynccontent/AsyncContent';
+import { editAppRequest } from './AppActionFactory';
+
+
+class EditApp extends React.Component {
+
+  static propTypes = {
+    actions: PropTypes.shape({
+      onEditApp: PropTypes.func.isRequired
+    }).isRequired,
+    editAppAsyncState: AsyncStatePropType.isRequired,
+    id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    description: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      name: '',
+      url: ''
+    };
+  }
+
+  componentDidMount() {
+  }
+
+  onTitleChange = (event) => {
+    this.setState({
+      title: event.target.value
+    });
+  };
+
+  onNameChange = (event) => {
+    this.setState({
+      name: event.target.value
+    });
+  };
+
+  onDescriptionChange = (event) => {
+    this.setState({
+      description: event.target.value
+    });
+  };
+
+  onUrlChange = (event) => {
+    this.setState({
+      url: event.target.value
+    });
+  };
+
+  onSubmit = () => {
+    const { title, description, name, url } = this.state;
+
+    // We need to collect only items that have been changed. AKA NOT an empty string.
+    // Having issues with this because of js and iterables.
+    // Collect the items, check each for change, add changed to appData object.
+    const tempMap = new Map([['title', title], ['description', description], ['name', name], ['url', url]]);
```

Comment:
This seems overly complicated. Why not just do it one by one...

```
const appData = {};

if (this.state.title && this.state.title.length > 0) {
  appData['title'] = this.state.title;
}

...

this.props.actions.onEditApp(appId, appData);
```

On a related note, we really should be using the model builders from `lattice-js`. I don't think there's models / model builders for App and AppType currently, but they should be created and used here.